datumaro/plugins/sampler/sampler.py

```diff
@@ -0,0 +1,212 @@
+            type=str,
+            help="Subset name to select sample",
+        )
+        parser.add_argument(
+            "-sample_name",
+            "--sampled_name",
+            type=str,
+            default="sampled_set",
+            help="sampled data subset name",
+        )
+        parser.add_argument(
+            "-unsample_name",
+            "--unsampled_name",
+            type=str,
+            default="unsampled_set",
+            help="unsampled data subset name name",
+        )
+        parser.add_argument(
+            "-m",
+            "--sampling_method",
+            type=str,
+            default="topk",
+            choices=["topk", "lowk", "randk", "mixk", "randtopk"],
+            help="Method of sampling, example: ['topk', 'lowk', 'randk', 'mixk', 'randtopk']",
+        )
+        parser.add_argument("-k", "--num_sample", type=int, help="Num of sample")
+        parser.add_argument(
+            "-o",
+            "--output_file",
+            type=str,
+            default=None,
+            help="Output Sample file path, The extension of the file must end with .csv",
+        )
+        return parser
+
+    def __init__(
+        self,
+        extractor,
+        algorithm,
+        subset_name,
+        sampled_name,
+        unsampled_name,
+        sampling_method,
+        num_sample,
+        output_file,
+    ):
+        """
+        Parameters
+        ----------
+        extractor : Extractor, Dataset
+        algorithm : str
+            Specifying the algorithm to calculate the uncertainty
+            for sample selection. default: 'entropy'
+        subset_name : str
+            The name of the subset to which you want to select a sample.
+        sample_name : str
+            Subset name of the selected sample, default: 'sample'
+        sampling_method : str
+            Method of sampling, 'topk' or 'lowk' or 'randk'
+        num_sample : int
+            Number of samples extracted
+        output_file : str
+            Path of sampler result, Use when user want to save results
+        """
+        super().__init__(extractor)
+
+        self.original_subset = subset_name
+        self.sampled_name = sampled_name
+        self.unsampled_name = unsampled_name
+
+        # optional. Use the --output_file option to save the sample list as a csv file.
+        if output_file is not None:
+            if output_file.split(".")[-1] != ".csv":
+                msg = f"Invalid extension, The extension of the file must end with .csv"
+                raise Exception(msg)
+
+        # 1. check the empty of the data set
+        if len(extractor) < 1:
+            msg = "Invalid Dataset, The dataset is empty."
+            raise Exception(msg)
+
+        # 2. Import data into a subset name and convert it
+        # to a format that will be used in the sampler algorithm with the inference result.
+        data_df, infer_df = self._load_inference_from_subset(extractor, subset_name)
```

Comment:
Please make the operation to be traversing the input dataset only when `__iter__` is called.